`tiup bench tpch` panic when it fails to access tidb

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

There is a tidb instance with user `test` and password properly set. I tried to run tpch benchmark test using following command and then tiup paniced and exited.

```
tiup bench tpch prepare --host 127.0.0.1 --port 4000 --db test --user test --password --sf 10                         17:31:47  tidb 
Starting component `bench`: /home/dylan/.tiup/components/bench/v1.5.6/tiup-bench tpch prepare --host 127.0.0.1 --port 4000 --db test --user test --password --sf 10
creating nation
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x558efce92404]

goroutine 100 [running]:
database/sql.(*Conn).close(0x0, 0x0, 0x0, 0x0, 0x0)
        database/sql/sql.go:1989 +0x44
database/sql.(*Conn).Close(...)
        database/sql/sql.go:2010
github.com/pingcap/go-tpc/tpch.Workloader.CleanupThread(0x0, 0x558efe8c06e0, 0xc0000d5b80, 0x558efe162000, 0xc000524960, 0x0)
        github.com/pingcap/go-tpc@v1.0.7/tpch/workload.go:98 +0x7f
panic(0x558efe000880, 0x558efe6f8a30)
        runtime/panic.go:965 +0x1b9
database/sql.(*Conn).grabConn(0x0, 0x558efe162000, 0xc000524960, 0xc0000a20d0, 0xc0004320c0, 0x10, 0xc0)
        database/sql/sql.go:1861 +0x28
database/sql.(*Conn).ExecContext(0x0, 0x558efe162000, 0xc000524960, 0x558efd789316, 0xc0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        database/sql/sql.go:1880 +0x4e
github.com/pingcap/go-tpc/tpch.(*Workloader).createTableDDL(0xc000161ba0, 0x558efe162000, 0xc000524960, 0x558efd789316, 0xc0, 0x558efd73ccbf, 0x6, 0x558efd73f7c8, 0x8, 0x0, ...)
        github.com/pingcap/go-tpc@v1.0.7/tpch/ddl.go:17 +0x1c5
github.com/pingcap/go-tpc/tpch.Workloader.createTables(0x0, 0x558efe8c06e0, 0xc0000d5b80, 0x558efe162000, 0xc000524960, 0x7f7abc683780, 0x0)
        github.com/pingcap/go-tpc@v1.0.7/tpch/ddl.go:41 +0x8a
github.com/pingcap/go-tpc/tpch.Workloader.Prepare(0x0, 0x558efe8c06e0, 0xc0000d5b80, 0x558efe162000, 0xc000524960, 0x0, 0x558efe162000, 0xc000524960)
        github.com/pingcap/go-tpc@v1.0.7/tpch/workload.go:106 +0x8e
main.execute(0x558efe161fc8, 0xc000524960, 0x558efe16bed0, 0xc00000cfd8, 0x558efd73e85a, 0x7, 0x0, 0x0, 0x0)
        github.com/tiup/components/bench/misc.go:51 +0x576
main.executeWorkload.func2(0xc00003ce70, 0x558efe161fc8, 0xc0000bd200, 0x558efe16bed0, 0xc00000cfd8, 0x558efd73e85a, 0x7, 0x0)
        github.com/tiup/components/bench/misc.go:104 +0xc5
created by main.executeWorkload
        github.com/tiup/components/bench/misc.go:102 +0x192
Error: run `/home/dylan/.tiup/components/bench/v1.5.6/tiup-bench` (wd:/home/dylan/.tiup/data/ShqR13d) failed: exit status 2
```

2. What did you expect to see?

Some clean and nice error message telling that no permission to access the tidb.

3. What did you see instead?

Rattled and busy stackstrace output due to the panic. 

4. What version of TiUP are you using (`tiup --version`)?

```
1.5.6 tiup
Go Version: go1.16.7
Git Ref: v1.5.6
GitHash: 1d1478dbec0d06ef5e909d3323331914538d4cb0
```

and

```tiup-bench v1.5.6```

--- a/components/bench/main.go
+++ b/components/bench/main.go
@@ -51,7 +51,7 @@ func closeDB() {
 	globalDB = nil
 }
 
-func openDB() {
+func openDB() error {
 	// TODO: support other drivers
 	var tmpDB *sql.DB
 	ds := fmt.Sprintf("%s:%s@tcp(%s:%d)/", user, password, host, port)
@@ -59,7 +59,7 @@ func openDB() {
 	var err error
 	globalDB, err = sql.Open(mysqlDriver, dsn)
 	if err != nil {
-		panic(err)
+		return err
 	}
 	if err := globalDB.Ping(); err != nil {
 		errString := err.Error()
@@ -67,14 +67,16 @@ func openDB() {
 			tmpDB, _ = sql.Open(mysqlDriver, ds)
 			defer tmpDB.Close()
 			if _, err := tmpDB.Exec(createDBDDL + dbName); err != nil {
-				panic(fmt.Errorf("failed to create database, err %v", err))
+				return fmt.Errorf("failed to create database, err %v", err)
 			}
 		} else {
 			globalDB = nil
+			return err
 		}
 	} else {
 		globalDB.SetMaxIdleConns(threads + 1)
 	}
+	return nil
 }
 
 func main() {

--- a/components/bench/tpcc.go
+++ b/components/bench/tpcc.go
@@ -26,7 +26,11 @@ func executeTpcc(action string) {
 	}
 	runtime.GOMAXPROCS(maxProcs)
 
-	openDB()
+	if err := openDB(); err != nil {
+		fmt.Println(err)
+		fmt.Println("Cannot open database, pleae check it (ip/port/username/password)")
+		os.Exit(1)
+	}
 
 	tpccConfig.DBName = dbName
 	tpccConfig.Threads = threads

--- a/components/bench/tpch.go
+++ b/components/bench/tpch.go
@@ -2,6 +2,8 @@ package main
 
 import (
 	"context"
+	"fmt"
+	"os"
 	"strings"
 
 	"github.com/pingcap/go-tpc/tpch"
@@ -11,7 +13,11 @@ import (
 var tpchConfig tpch.Config
 
 func executeTpch(action string, _ []string) {
-	openDB()
+	if err := openDB(); err != nil {
+		fmt.Println(err)
+		fmt.Println("Cannot open database, pleae check it (ip/port/username/password)")
+		os.Exit(1)
+	}
 	defer closeDB()
 
 	tpchConfig.DBName = dbName

